server/src/main/scala/org/http4s/server/armeria/ArmeriaServerBuilder.scala

```diff
@@ -0,0 +1,334 @@
+    * `keyFile` and `keyPassword`.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(keyCertChainFile: File, keyFile: File, keyPassword: Option[String]): Self = {
+    armeriaServerBuilder.tls(keyCertChainFile, keyFile, keyPassword.orNull)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] with the specified `keyCertChainInputStream`,
+    * `keyInputStream` and `keyPassword`.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(
+      keyCertChainInputStream: InputStream,
+      keyInputStream: InputStream,
+      keyPassword: Option[String]): Self = {
+    armeriaServerBuilder.tls(keyCertChainInputStream, keyInputStream, keyPassword.orNull)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] with the specified cleartext [[PrivateKey]] and
+    * [[X509Certificate]] chain.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(key: PrivateKey, keyCertChain: X509Certificate*): Self = {
+    armeriaServerBuilder.tls(key, keyCertChain: _*)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] with the specified [[KeyManagerFactory]].
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(keyManagerFactory: KeyManagerFactory): Self = {
+    armeriaServerBuilder.tls(keyManagerFactory)
+    this
+  }
+
+  /** Configures SSL or TLS of the [[BackendServer]] with an auto-generated self-signed certificate.
+    * '''Note:''' You should never use this in production but only for a testing purpose.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTlsSelfSigned: Self = {
+    armeriaServerBuilder.tlsSelfSigned
+    this
+  }
+
+  /** Adds the specified `tlsCustomizer` which can arbitrarily configure the [[SslContextBuilder]] that will be
+    * applied to the SSL session.
+    */
+  def withTlsCustomizer(tlsCustomizer: SslContextBuilder => Unit): Self = {
+    armeriaServerBuilder.tlsCustomizer(ctxBuilder => tlsCustomizer(ctxBuilder))
+    this
+  }
+
+  /** Sets the amount of time to wait after calling [[BackendServer#stop()]] for
+    * requests to go away before actually shutting down.
+    *
+    * @param quietPeriod the number of milliseconds to wait for active
+    *                    requests to go end before shutting down. [[Duration.Zero]] means
+    *                    the server will stop right away without waiting.
+    * @param timeout     the amount of time to wait before shutting down the server regardless of active
+    *                    requests.
+    *                    This should be set to a time greater than `quietPeriod` to ensure the server
+    *                    shuts down even if there is a stuck request.
+    */
+  def withGracefulShutdownTimeout(quietPeriod: Duration, timeout: Duration): Self = {
+    armeriaServerBuilder.gracefulShutdownTimeoutMillis(quietPeriod.toMillis, timeout.toMillis)
```

Comment:
This will crash if `InfiniteDuration`s are provided.  I would check those, or require `FiniteDuration`.

Reply:
That's a good suggestion. Let me check `InfiniteDuration`s too.